src/datasets/builder.py

```diff
@@ -331,6 +336,7 @@ def __init__(
         self.use_auth_token = use_auth_token
         self.repo_id = repo_id
         self.storage_options = storage_options
+        self.dataset_name = camelcase_to_snakecase(dataset_name) if dataset_name else self.name
```

Comment:
did this in case name in camel case format for some reason is passed. but the way it's currently used this formatting is done before passing to builder (in `load.py`). so maybe i can drop it, but we do this for `self.name` here several lines above so I'd actually leave it for consistency (`dataset_name` is used for cache folder name so it must be snake-cased)

Reply:
Ah right - it was already set to camel case for datasets with scripts !